src/main/scala/uk/gov/nationalarchives/referencegenerator/Lambda.scala

```diff
@@ -1,12 +1,58 @@
 package uk.gov.nationalarchives.referencegenerator
 
-import uk.gov.nationalarchives.referencegenerator.Lambda.Input
+import com.amazonaws.services.lambda.runtime.events.{APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent}
+import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
+import com.typesafe.config.{Config, ConfigFactory}
+import com.typesafe.scalalogging.Logger
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+import uk.gov.nationalarchives.referencegenerator.Lambda.{EncryptedReference, Input}
+import io.circe.syntax._
 
-class Lambda {
+import scala.util.{Failure, Success}
 
-  def process(input: Input): Unit = { }
+class Lambda extends RequestHandler[APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent] {
+  val logger: Logger = Logger[Lambda]
+  val client: DynamoDbClient = DynamoDbClient.builder()
+    .credentialsProvider(DefaultCredentialsProvider.create())
+    .region(Region.EU_WEST_2)
+    .build()
+
+  override def handleRequest(event: APIGatewayProxyRequestEvent, context: Context): APIGatewayProxyResponseEvent = {
+    val config: Config = ConfigFactory.load()
+    val queryParams = event.getQueryStringParameters
+    val queryParam: String = config.getString("dynamodb.queryParam")
+    val numberOfRefs = queryParams.get(queryParam).toInt
+
+    process(Input(numberOfRefs), client)
+  }
+
+  def process(input: Input, client: DynamoDbClient = client): APIGatewayProxyResponseEvent = {
```

Comment:
Instead of passing the "client" into the method it can be passed in as a constructor param to the lambda class itself, and use the companion object to instantiate the class using the `apply` method. This will still allow mocking in the tests, but tidies out the code and reduces the amount of passing around the "client" does. 

Something like this:

```
class Lambda(counterClient: DynamoDbClient) extends RequestHandler[APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent] {
 ...  etc ...
}

object Lambda {

  val counterClient: DynamoDbClient = DynamoDbClient.builder()
        ... etc ...
    .build()

  def apply(): Lambda = new Lambda(counterClient)
  ... etc ...
}
```

Can also handle the "Config" in the sameway, passing it into the class constructor via the companion object.

Reply:
Good idea. I've moved these client and config to the companion object